libbeat/docs/outputconfig.asciidoc

```diff
@@ -86,13 +86,85 @@ output.elasticsearch:
   protocol: "https"
   username: "{beatname_lc}_internal"
   password: "{pwd}"
-
 ------------------------------------------------------------------------------
 
 
 For more information about securing {beatname_uc}, see
 <<securing-{beatname_lc}>>.
 
+
+To configure the Elasticsearch output with https://www.elastic.co/guide/en/elasticsearch/reference/6.x/getting-started-index-lifecycle-management.html[index lifecycle management] feature configure it as following:
+
+[source,yaml]
+------------------------------------------------------------------------------
+output.elasticsearch:
+  hosts: ["localhost:9200"]
+  ilm.enabled: true
+------------------------------------------------------------------------------
+
+This will overwrite your index settings and adjust the Beats template to contain the correct settings.
+
+Note: If you have loaded a template before into Elasticsearch with the same version you must overwrite the template.
+Note: This feature only works with X-Pack Basic.
+
+The default `rollover_alias` used is `{beatname_lc}-{beat.version}`. It's not possible to remove the `{beat.version}`
+from the rollover_alias name but the prefix part can be adjusted. This will also directly affect the name of the
+rollover indices. The default pattern used for the rollover index is `%7Bnow%2Fd%7D-000001`. It can be adjusted with the
+`ilm.pattern` config. See config below for an example:
+
+[source,yaml]
+------------------------------------------------------------------------------
+output.elasticsearch:
+  hosts: ["localhost"]
+  ilm.enabled: true
+  ilm.rollover_alias: "metricbeat"
+  ilm.pattern: "%7Bnow%2Fd%7D-000001"
+------------------------------------------------------------------------------
+
+Note: If you modify the rollover_alias or the pattern you have to overwrite the index template again as the settings
+`index.lifecycle.name` and `index.lifecycle.rollover_alias` change automatically in the template.
+
+For ilm pattern date math can be used, see https://www.elastic.co/guide/en/elasticsearch/reference/master/indices-rollover-index.html#_using_date_math_with_the_rollover_api[the rollover docs] for more details.
```

Comment:
I’d probably say it as 

“Data math is supported in the `ilm.pattern`...”
or
“Date math is supported by the ILM pattern...”